lxd/patches.go

```diff
@@ -2130,3 +2131,94 @@ func patchStorageApiLxdOnBtrfs(name string, d *Daemon) error {
 
 	return nil
 }
+
+func patchStorageApiDetectLVSize(name string, d *Daemon) error {
+	pools, err := dbStoragePools(d.db)
+	if err != nil {
+		if err == NoSuchObjectError {
+			return nil
+		}
+		shared.LogErrorf("Failed to query database: %s", err)
+		return err
+	}
+
+	for _, poolName := range pools {
+		poolID, pool, err := dbStoragePoolGet(d.db, poolName)
+		if err != nil {
+			shared.LogErrorf("Failed to query database: %s", err)
+			return err
+		}
+
+		// Make sure that config is not empty.
+		if pool.Config == nil {
+			pool.Config = map[string]string{}
+
+			// Insert default values.
+			err = storagePoolFillDefault(poolName, pool.Driver, pool.Config)
+			if err != nil {
+				return err
+			}
+		}
+
+		// We're only interested in LVM pools.
+		if pool.Driver != "lvm" {
+			continue
+		}
+
+		// Get all storage volumes on the storage pool.
+		volumes, err := dbStoragePoolVolumesGet(d.db, poolID, supportedVolumeTypes)
+		if err != nil {
+			if err == NoSuchObjectError {
+				continue
+			}
+			return err
+		}
+
+		poolName := pool.Config["lvm.vg_name"]
+		if poolName == "" {
+			shared.LogErrorf("The \"lvm.vg_name\" key should not be empty.")
+			return fmt.Errorf("The \"lvm.vg_name\" key should not be empty.")
+		}
+
+		for _, volume := range volumes {
+			// Make sure that config is not empty.
+			if volume.Config == nil {
+				volume.Config = map[string]string{}
+
+				// Insert default values.
+				err := storageVolumeFillDefault(volume.Name, volume.Config, pool)
+				if err != nil {
+					return err
+				}
+			}
+
+			// It shouldn't be possible that false volume types
+			// exist in the db, so it's safe to ignore the error.
+			volumeTypeApiEndpoint, _ := storagePoolVolumeTypeNameToApiEndpoint(volume.Type)
+			lvmName := containerNameToLVName(volume.Name)
```

Comment:
Using `containerNameToLVName()` unconditionally is safe since `images` and the like cannot contain `-` afaict, so we are fine. Should also rename that function to `getValidLVName()`.